testrepository/commands/failing.py

```diff
@@ -79,8 +96,10 @@ def run(self):
             result = 1
         else:
             result = 0
-        if self.ui.options.list:
-            failing_tests = [
-                test for test, _ in summary.errors + summary.failures]
-            self.ui.output_tests(failing_tests)
+        if self.ui.options.list or self.ui.options.json:
+            if self.ui.options.json:
+                style = 'json'
+            else:
+                style = 'list'
```

Comment:
This "if it's list or json" thing shows up a few times, which makes me think that there's maybe some object-y goodness that could be extracted.